common/src/hand_off.rs

```diff
@@ -374,11 +375,30 @@ pub fn print_fht(fht: &FirmwareHandoffTable) {
 }
 
 impl FirmwareHandoffTable {
+    fn reset_reason() -> ResetReason {
```

Comment:
I don't see this as part of the set of operations for the FHT type. We can make it a free funtion.